features/enlinkd/tests/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdTestHelper.java

```diff
@@ -3713,47 +3676,47 @@ public void check(BroadcastDomain domain) throws BridgeTopologyException {
             for (SharedSegment shared: links) {
-                assertTrue(!shared.noMacsOnSegment());
+                assertFalse(shared.noMacsOnSegment());
                 assertEquals(nodeAId,shared.getDesignatedBridge());
                 assertEquals(1, shared.getBridgeIdsOnSegment().size());
                 assertEquals(0, SharedSegment.getBridgeBridgeLinks(shared).size());
                 for (BridgeMacLink link: SharedSegment.getBridgeMacLinks(shared)) {
                     assertEquals(link.getBridgePort(),shared.getDesignatedPort().getBridgePort());
-                    if (link.getBridgePort() == portA1) {
+                    if (Objects.equals(link.getBridgePort(), portA1)) {
                         assertEquals(1, shared.getMacsOnSegment().size());
                         assertEquals(mac1, link.getMacAddress());
-                    } else if (link.getBridgePort() == portA2) {
+                    } else if (Objects.equals(link.getBridgePort(), portA2)) {
                         assertEquals(1, shared.getMacsOnSegment().size());
                         assertEquals(mac2, link.getMacAddress());
-                    } else if (link.getBridgePort() == portA3) {
+                    } else if (Objects.equals(link.getBridgePort(), portA3)) {
                         assertEquals(1, shared.getMacsOnSegment().size());
                         assertEquals(mac3, link.getMacAddress());
-                    } else if (link.getBridgePort() == portA4) {
+                    } else if (Objects.equals(link.getBridgePort(), portA4)) {
                         assertEquals(1, shared.getMacsOnSegment().size());
                         assertEquals(mac4, link.getMacAddress());
-                    } else if (link.getBridgePort() == portA23) {
+                    } else if (Objects.equals(link.getBridgePort(), portA23)) {
                         final Set<String> macs = shared.getMacsOnSegment();
                         assertEquals(4, macs.size());
                         assertTrue(macs.contains(mac231));
                         assertTrue(macs.contains(mac232));
                         assertTrue(macs.contains(mac233));
                         assertTrue(macs.contains(mac234));
-                    } else if (link.getBridgePort() == portA24) {
+                    } else if (Objects.equals(link.getBridgePort(), portA24)) {
                         final Set<String> macs = shared.getMacsOnSegment();
                         assertEquals(5, macs.size());
                         assertTrue(macs.contains(mac241));
                         assertTrue(macs.contains(mac242));
                         assertTrue(macs.contains(mac243));
                         assertTrue(macs.contains(mac244));
                         assertTrue(macs.contains(mac245));
-                    } else if (link.getBridgePort() == portA25) {
+                    } else if (Objects.equals(link.getBridgePort(), portA25)) {
```

Comment:
nit: could also be portA25.equals(link.getBridgePort())

Reply:
eheh right